main_test.go

```diff
@@ -193,32 +198,85 @@ func runTest(t *testing.T, test *mainTest) {
 
 	input := read(test.input)
 
+	eKind := evalKindRegular
+	// If the golden path is to a directory, this is a multi-test.
+	if info, err := os.Stat(test.golden); err == nil && info.IsDir() {
+		eKind = evalKindMulti
+	}
+
 	result := runJsonnet(jsonnetInput{
-		name:    test.name,
-		input:   input,
-		eKind:   evalKindRegular,
-		extVars: test.meta.extVars,
-		extCode: test.meta.extCode,
+		name:         test.name,
+		input:        input,
+		eKind:        eKind,
+		stringOutput: strings.HasSuffix(test.golden, "_string_output.golden"),
+		extVars:      test.meta.extVars,
+		extCode:      test.meta.extCode,
 	})
 
-	// TODO(sbarzowski) report which files were updated
-	if *update {
-		err := ioutil.WriteFile(test.golden, []byte(result.output), 0666)
-		if err != nil {
-			t.Errorf("error updating golden files: %v", err)
+	compareGolden := func(inputPath, goldenPath, result string, golden []byte) {
+		if bytes.Compare(golden, []byte(result)) != 0 {
+			// TODO(sbarzowski) better reporting of differences in whitespace
+			// missing newline issues can be very subtle now
+			t.Fail()
+			t.Errorf("Mismatch when running %s. Golden: %s\n", inputPath, goldenPath)
+			data := diff(result, string(golden))
+			t.Errorf("diff %s jsonnet %s\n", goldenPath, inputPath)
+			t.Errorf(string(data))
 		}
-		return
 	}
-	golden := read(test.golden)
-	if bytes.Compare(golden, []byte(result.output)) != 0 {
-		// TODO(sbarzowski) better reporting of differences in whitespace
-		// missing newline issues can be very subtle now
-		t.Fail()
-		t.Errorf("Mismatch when running %s.jsonnet. Golden: %s\n", test.name, test.golden)
-		data := diff(result.output, string(golden))
-		t.Errorf("diff %s jsonnet %s.jsonnet\n", test.golden, test.name)
-		t.Errorf(string(data))
 
+	switch eKind {
```

Comment:
I did something similar, but I made the single/multifile function signatures compatible so that there's only a "single" codepath for doing the diff. WDYT?